chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/impl/ComplaintDAOImpl.java

```diff
@@ -0,0 +1,258 @@
+        } else {
+            entity.setComplaintType(getComplaintTypeEntityById(complaintDTO.getComplaintType().getId()));
+        }
+
+        entity.setOncComplaintId(complaintDTO.getOncComplaintId());
+        entity.setAcbComplaintId(complaintDTO.getAcbComplaintId());
+        entity.setReceivedDate(complaintDTO.getReceivedDate());
+        entity.setSummary(complaintDTO.getSummary());
+        entity.setActions(complaintDTO.getActions());
+        entity.setComplainantContacted(complaintDTO.isComplainantContacted());
+        entity.setDeveloperContacted(complaintDTO.isDeveloperContacted());
+        entity.setOncAtlContacted(complaintDTO.isOncAtlContacted());
+        entity.setClosedDate(complaintDTO.getClosedDate());
+
+        entity.setDeleted(false);
+        entity.setLastModifiedUser(AuthUtil.getAuditId());
+
+        entityManager.merge(entity);
+        entityManager.flush();
+
+        ComplaintEntity updatedEntity = getEntityById(complaintDTO.getId());
+        return new ComplaintDTO(updatedEntity);
+    }
+
+    @Override
+    public void delete(ComplaintDTO complaintDTO) throws EntityRetrievalException {
+        ComplaintEntity entity = getEntityById(complaintDTO.getId());
+        entity.setDeleted(true);
+        entity.setLastModifiedUser(AuthUtil.getAuditId());
+
+        entityManager.merge(entity);
+        entityManager.flush();
+    }
+
+    private ComplaintEntity getEntityById(Long id) throws EntityRetrievalException {
+        ComplaintEntity entity = null;
+
+        Query query = entityManager.createQuery("FROM ComplaintEntity c " + "JOIN FETCH c.certificationBody "
+                + "LEFT JOIN FETCH c.complaintType " + "LEFT JOIN FETCH c.complaintStatusType "
+                + "WHERE c.deleted = false " + "AND c.id = :complaintId", ComplaintEntity.class);
+        query.setParameter("complaintId", id);
+        List<ComplaintEntity> result = query.getResultList();
+
+        if (result.size() > 1) {
+            throw new EntityRetrievalException("Data error. Duplicate filter id in database.");
+        } else if (result.size() == 1) {
+            entity = result.get(0);
+        }
+        return entity;
+    }
+
+    private List<ComplaintTypeEntity> getComplaintTypeEntities() {
+        Query query = entityManager.createQuery("from ComplaintTypeEntity where (NOT deleted = true) ",
+                ComplaintTypeEntity.class);
+        List<ComplaintTypeEntity> result = query.getResultList();
+        return result;
+    }
+
+    private List<ComplaintStatusTypeEntity> getComplaintStatusTypeEntities() {
+        Query query = entityManager.createQuery("from ComplaintStatusTypeEntity where (NOT deleted = true) ",
+                ComplaintStatusTypeEntity.class);
+        List<ComplaintStatusTypeEntity> result = query.getResultList();
+        return result;
+    }
+
+    private CertificationBodyEntity getAcbEntityById(final Long entityId) throws EntityRetrievalException {
```

Comment:
If you change the entity as mentioned, you would only need the ACB ID and wouldn't need this method.

Reply:
Based on the comment above, I'll remove this method.